tests/e2e/helpers/runners.py

```diff
@@ -197,12 +197,18 @@ def _run_once(
             finally:
                 chunk = _get_chunk(child)
                 output += chunk
-        if allow_nonzero_exitcode:
-            child.close()
+        if assert_exit_code:
+            if child.isalive():
+                # flush process buffer
+                child.read()
+                # wait for child to exit
+                log_msg(f"Waiting for {cmd}", logger=LOGGER.info)
+                child.wait()
+            child.close(force=True)
```

Comment:
thank you, it seems to work! the last thing: let's add a few doctests here covering following situation:
1. `cmd="echo 1; sleep 2; echo 2", expected_patterns=["1"]`, so we check that the job's exit code is `0` and its output contains `2`
2. same `cmd`, `expected_patterns=[]`
3. `cmd="echo 1"` -- we check the branch where `child.isalive()` is False

Reply:
@anayden can we have these changes tested?

Reply:
We need this functionality. So let's merge now and add tests later: https://github.com/neuromation/cookiecutter-neuro-project/issues/230